### DISCUSSION TITLE
Lab 02.02 task 1 Solucionado

### DISCUSSION THREAD

**Discussion by Astolfo2332** [rramosp/ai4eng.v1#215] [02 - PYTHON]
Hola tengo un problema con el evaluador ya que aunque coloque la excepción de ValueError me sigue colocando el punto malo.
```
def cauchy(x, y):
    r=np.zeros((np.shape(x)[0],np.shape(y)[0]))
    con_i,con_j=0,0
    for i in x:
        for j in y:
            try:
                r[con_i][con_j]=1/(i-j)
            except ValueError:
                r[con_i][con_j]=0
            con_j+=1
        con_i+=1
        con_j=0
    return r
```
Lo cual funciona de maravilla, pero en el caso de la división de 0 no me causa un error si no un warning:

```
<ipython-input-2-06b159f115f7>:8: RuntimeWarning: divide by zero encountered in long_scalars
  r[con_i][con_j]=1/(i-j)
array([[ 0.03571429,  0.02631579,  0.03333333, -0.03333333,  0.03703704],
       [ 0.07142857,  0.04166667,  0.0625    , -0.02272727,  0.07692308],
       [ 0.02      ,  0.01666667,  0.01923077, -0.125     ,  0.02040816],
       [ 0.01724138,  0.01470588,  0.01666667,         inf,  0.01754386],
       [ 0.02702703,  0.0212766 ,  0.02564103, -0.04761905,  0.02777778]])
```

Y cuando lo subo me causa:

```
task_01 submitted. your grade is 0
----- grader message -------
incorrect: testing with 100 random calls
your code did not correctly detect a division by zero
x = [84 44 24 79 39 34 63 92], y=[ 1 1 6 1 39 3 13 3]
```
Que justamente es el error al cual deberia hacer la excepción.

**Comment by Astolfo2332** [comment]
Al final lo soucione caturando el ValueError manualmente como sugeria el profesor [aqui](https://github.com/rramosp/ai4eng.v1/discussions/35#discussioncomment-911723), por si alguien le sirve :

```
if i-j==0:
       raise ValueError("No se admite división por 0")
else:
      r[con_i][con_j]=1/(i-j)
       con_j+=1
```

**Comment by rramosp** [comment]
gracias Miguel .. .efectivamente el ejercicio requeria que **lanzaras** las
excepciòn con raise ... no simplemente que la capturaras

On Fri, Feb 10, 2023 at 7:56 AM Miguel ***@***.***> wrote:

> Al final lo soucione caturando el ValueError manualmente como sugeria el
> profesor aqui
> <https://github.com/rramosp/ai4eng.v1/discussions/35#discussioncomment-911723>,
> por si alguien le sirve :
>
> if i-j==0:
>
>        raise ValueError("No se admite división por 0")
>
> else:
>
>       r[con_i][con_j]=1/(i-j)
>
>        con_j+=1
>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/215#discussioncomment-4934840>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIPOLEESYREBY7AC2E3WWY3JRANCNFSM6AAAAAAUXA74FI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.
